### DISCUSSION TITLE
Purpose of bootloader for ATTiny1614

### DISCUSSION THREAD

**Discussion by ghost** [SpenceKonde/megaTinyCore#1066] [Q&A] [answered]
Is there any point burning a bootloader to an ATTiny1614 if I am using UPDI to program the part? Someone on YouTube asserted that unless a bootloader is added then some of the settings enabled by megaTinyCore in the Arduino IDE would not be loaded to the target device, resulting in unpredictable results, I find this hard to believe, does anyone have a definitive answer?

**Comment by hmeijdam** [comment, marked as answer]
If you want to set the BOD mode or voltage level, save or erase EEPROM or if you want to set the WDT as mandatory you need to give the command "Burn Bootloader". 
But this also works when you select the board without optiboot.

So you don't need a bootloader (optiboot), but you do need to give the command "Burn Bootloader" to activate/manipulate those features.

This is why you see (burn bootloader req'd) in the menu options.
![image](https://github.com/SpenceKonde/megaTinyCore/assets/9595465/0adf592f-c3b4-4a09-af68-36db990a2e32)

**Comment by ghost** [comment]
OK I see, so should I burn the bootloader whenever I change any of the settings that mandate burn bootloader req'd?  I assume that burn bootloader does not install Optiboot if I haven't selected this board option?  What happens if burn bootloader is not used, do the various settings have default values, unless overwritten by burn bootloader?

**Reply by hmeijdam** [reply]
That's correct. 
In chapter 6.10.4 of the datasheet you can see the default (reset) values of the fuses
Here is the BOD configuration fuse register
![image](https://github.com/SpenceKonde/megaTinyCore/assets/9595465/a4118aa6-8037-45cc-aa90-1e3a96461fd7)

**Reply by ghost** [reply]
OK so I assume the BOD Configuration fuse would be 0x0 on a new part i.e. BOD = 1.8v, or is it undefined? I can then change the fuse settings to e.g. 0x7 or 4.2v using the 'burn bootloader' operation and those fuses would remain at 0x7 until I changed them again using 'burn bootloader' even after a reset, or power cycle?  The fuse settings are then transferred to the BOD Configuration Register at every start-up.  So is the potential problem that some fuse settings are, or may be, undefined on a new part so it is essential to set them to a known value using 'burn bootloader'?

**Reply by hmeijdam** [reply]
The level is at 1.8V , but bit 3 and 2 are also zero, which means BOD is disabled. You would have to set bit 2 to "1" to enable BOD during "active state"  if you also want BOD during sleep you need to set bit "0" as well.

The parts are factory shipped with the RESET value.

**Reply by ghost** [reply]
So is this the correct sequence?
1: Change BOD voltage in Arduino IDE
2: Use 'burn bootloader' to change BOD fuses on MCU
3: Upload sketch using UPDI
4: UPDI upload applies this software reset process to make the fuse change effective, as described in 33.5.6 ASI Reset Request:
...................pymcuprog.serialupdi.application - INFO - Apply reset
...................pymcuprog.serialupdi.link - INFO - STCS 59 to 0x08
...................pymcuprog.serialupdi.application - INFO - Release reset
...................pymcuprog.serialupdi.link - INFO - STCS 00 to 0x08
5: BOD fuses transferred to BOD Config Register at startup
6: Normal sketch execution

**Reply by hmeijdam** [reply]
That's how I do it. I look in the datasheet which BOD level is applicable for my clock speed and then do burn bootloader once and use UPDI for the rest. I have literally never used a bootloader on a modern Attiny. I only see disadvantages.

**Reply by ghost** [reply]
Thanks for your help Hans.
